internals/webhooks/validate/subnamespace/subnamespace_update.go

```diff
@@ -0,0 +1,197 @@
+package webhooks
+
+import (
+	"fmt"
+	"github.com/dana-team/hns/internals/utils"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// handleUpdate implements the non-boilerplate logic of the validator, allowing it to be more easily unit
+// tested (i.e. without constructing a full admission.Request)
+func (a *SubnamespaceAnnotator) handleUpdate(snsObject, snsOldObject *utils.ObjectContext) admission.Response {
+	if rsp := a.validateRPLabelDeletion(snsObject, snsOldObject); !rsp.Allowed {
+		return rsp
+	}
+
+	snsParentName := snsObject.Object.GetNamespace()
+	snsParentNS, err := utils.NewObjectContext(snsObject.Ctx, snsObject.Client, types.NamespacedName{Name: snsParentName}, &corev1.Namespace{})
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	isSNSResourcePool, err := utils.IsSNSResourcePool(snsObject.Object)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	isOldSNSResourcePool, err := utils.IsSNSResourcePool(snsOldObject.Object)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	isParentNSResourcePool, err := utils.IsNamespaceResourcePool(snsParentNS)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if rsp := a.validateRPToSubnamespace(snsObject, isSNSResourcePool, isOldSNSResourcePool, isParentNSResourcePool); !rsp.Allowed {
+		return rsp
+	}
+
+	if !isOldSNSResourcePool {
+		if rsp := a.validateResourceQuotaParams(snsObject, isSNSResourcePool); !rsp.Allowed {
+			return rsp
+		}
+	}
+
+	if rsp := a.validateSNSQuotaObject(snsObject); !rsp.Allowed {
+		return admission.Allowed("")
+	}
+
+	if rsp := a.validateEnoughResourcesForChildren(snsObject); !rsp.Allowed {
+		return rsp
+	}
+
+	// validate the request if the subnamespace is not a ResourcePool OR the subnamespace is the upper ResourcePool
+	// this is because only in that case the subnamespace would have a RQ or CRQ attached to it.
+	// Otherwise, the subnamespace is part of a ResourcePool (and does not have a RQ/CRQ attached to it) and this check is unneeded
+	isSNSUpperResourcePool, err := utils.IsSNSUpperResourcePool(snsObject)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if !isSNSResourcePool || isSNSUpperResourcePool {
+		if rsp := a.validateUpdateSnsRequest(snsObject, snsOldObject); !rsp.Allowed {
+			return rsp
+		}
+	}
+
+	return admission.Allowed("")
+}
+
+// validateRPLabelDeletion validates that the ResourcePool label has not been deleted
+func (a *SubnamespaceAnnotator) validateRPLabelDeletion(snsObject, snsOldObject *utils.ObjectContext) admission.Response {
+	snsResourcePoolLabel := utils.GetSNSResourcePoolLabel(snsObject.Object)
+	oldSNSResourcePoolLabel := utils.GetSNSResourcePoolLabel(snsOldObject.Object)
+
+	if snsResourcePoolLabel == "" && oldSNSResourcePoolLabel != "" {
```

Comment:
"" is equal to false so this if could look like this
```suggestion
	if !snsResourcePoolLabel  && oldSNSResourcePoolLabel  {
```